src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.ios.cs

```diff
@@ -69,43 +67,45 @@ void ClearTintColor()
 			}
 		}
 
-		async void SetUIImageViewTintColor(UIImageView imageView, Color color)
+		void SetUIImageViewTintColor(UIImageView imageView, Color color)
 		{
 			if (imageView.Image == null)
 			{
-				if (_retryCount < 4)
-				{
-					_retryCount++;
-					await Task.Delay(100);
-					SetUIImageViewTintColor(imageView, color);
-				}
-				else
+				Element.PropertyChanged += (_, e) =>
```

Comment:
We are almost there; I suggest creating a method for this because we need to unsubscribe this event when the effect is Detached (same for ImageButton)